### PR TITLE
feat: add github actions pipelines

### PR DESCRIPTION
**Pull request by f0rmiga**
Adds a Github Actions pipeline that uses Bazel remote cache and remote build execution.
The way to trigger CI is via the `ci-test` label. This avoids overloading the remote cache and remote build executors.

### CODE REVIEW

File: .github/workflows/default.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
+name: Default
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    types: [labeled]
+
+  # Allows manually triggering this workflow from the Actions tab.
+  workflow_dispatch:
```
Comment (kormide) [inline .github/workflows/default.yaml:10]: This doesn't need a value?

Comment (f0rmiga) [reply .github/workflows/default.yaml:10]: No

File: .github/workflows/default.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
+name: Default
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    types: [labeled]
```
Comment (kormide) [inline .github/workflows/default.yaml:7]: Only for labelled PRs?

Comment (f0rmiga) [reply .github/workflows/default.yaml:7]: Yes, I tried to explain it in the PR description:
```The way to trigger CI is via the ci-test label. This avoids overloading the remote cache and remote build executors.```

Want me to add as a comment?

Comment (kormide) [reply .github/workflows/default.yaml:7]: Nope, just curious.

**Pull request merged**